### DISCUSSION TITLE
Ability to view/edit arbitrary config files from UI (like .env file)

### DISCUSSION THREAD

**Discussion by semiligneous** [louislam/dockge#639] [General]
Before raising as a feature request, I wanted to discuss whether this has been considered previously, and perhaps whether was already determined to be a bad idea / in-plausible for some reason.

The ability to edit the .env file is great, but many containers have other files providing configuration (and they may not be able to use .env since it is more constrained in it's format). Examples include Caddyfile for Caddy, config.yaml for Immich-Kiosk, nginx.conf for nginx (as per [this discussion)](https://github.com/louislam/dockge/discussions/608).

I love that dockge allows me to administer my stacks without needing to get into a terminal, making it easy from mobile and other devices around the home. However, many containers require changes to other config files, and then I have to get to a terminal anyway.

I suspect the arbitrary file would be defined for each container - presumably in a dockge config/.env file such as:

`custom_config_file_for_[container-name]: path/to/config.file`

so an example of `dockge/.env` might have:

```
custom_config_file_for_caddy: Caddyfile
custom_config_file_for_nginx-web-server-1: nginx.conf
custom_config_file_for_immich-kiosk: config.yaml
```

I'm sure there's more to consider! Would be great to discuss.

thanks

**Comment by nsideguy** [comment]
I would also like a way to edit files besides the compose and .env file. Many containers have no use for an .env file as well

**Reply by scelfo** [reply]
This would be a fantastic addition! My specific use case is very similar to what @semiligneous mentions. I want to be able to edit a tailscale serve.config like what's described on https://tailscale.com/blog/docker-tailscale-guide.

**Comment by nvlbravo** [comment]
Agreed - I was just looking for a way to add other `.env` files.

Based on this (https://github.com/louislam/dockge/blob/master/CONTRIBUTING.md?plain=1#L41), I don't think an `.env` file-based approach for configuration would be merged.

What about replacing or adding something below the default "ENV File" section with something to allow you to configure add-on files as needed, such as:
![image](https://github.com/user-attachments/assets/9c577019-e7a4-41b6-8882-bf5d60f476db)
And out pops a:
![image](https://github.com/user-attachments/assets/f69b9722-601c-430e-9626-aa80b5f61f14)

Seems like modifying the frontend might be replacing this bit with controls for the above and displaying an `editor-box` for each custom file: https://github.com/louislam/dockge/blob/master/frontend/src/pages/Compose.vue#L185
Backend would likely be replacing or adding to this bit and all the related methods to accept a dynamic list of values: https://github.com/louislam/dockge/blob/master/backend/stack.ts#L28
